src/applications/personalization/appointments/actions/index.js

```diff
@@ -1,27 +1,216 @@
+import MOCK_CC_APPOINTMENTS from '~/applications/personalization/dashboard-2/utils/mocks/appointments/MOCK_CC_APPOINTMENTS';
 
-export const FETCH_APPOINTMENTS = 'FETCH_APPOINTMENTS';
-export const FETCH_APPOINTMENTS_SUCCESS = 'FETCH_APPOINTMENTS_SUCCESS';
-export const FETCH_APPOINTMENTS_FAILURE = 'FETCH_APPOINTMENTS_FAILURE';
+function isAtlasLocation(appointment) {
+  return appointment.attributes?.vvsAppointments?.some(
+    element => element.tasInfo,
+  );
+}
+
+function getAdditionalInfo(appointment) {
+  const appointmentKind =
+    appointment.attributes?.vvsAppointments?.[0]?.appointmentKind;
+
+  if (isAtlasLocation(appointment)) {
+    return 'at an ATLAS location';
+  }
+
+  if (appointmentKind === VIDEO_TYPES.clinic) {
+    return 'at a VA location';
+  }
+
+  if (appointmentKind === VIDEO_TYPES.gfe) {
+    return 'using a VA device';
+  }
+
+  if (
+    appointmentKind === VIDEO_TYPES.adhoc ||
+    appointmentKind === VIDEO_TYPES.mobile
+  ) {
+    return 'at home';
+  }
+
+  return null;
+}
+
+function getStagingID(facilityID) {
+  if (!facilityID) {
+    return facilityID;
+  }
+
+  if (environment.isProduction()) {
+    return facilityID;
+  }
 
-export function fetchAppointments() {
-  return dispatch => {
+  if (facilityID.startsWith('983')) {
+    return facilityID.replace('983', '442');
+  }
+
+  if (facilityID.startsWith('984')) {
+    return facilityID.replace('984', '552');
+  }
+
+  return facilityID;
+}
+
+export function fetchConfirmedFutureAppointments() {
+  return async dispatch => {
     dispatch({
-      type: FETCH_APPOINTMENTS,
+      type: FETCH_CONFIRMED_FUTURE_APPOINTMENTS,
     });
 
-    return apiRequest('/appointments')
-      .then(res =>
-        dispatch({
-          type: FETCH_APPOINTMENTS_SUCCESS,
-          data: res.data.attributes.appointments,
-        }),
-      )
-      .catch(err => {
-        dispatch({
-          type: FETCH_APPOINTMENTS_FAILURE,
-          err,
-        });
+    let facilitiesLookup = {};
+    let facilitiesResponse;
+    let vaAppointments = [];
+    let ccAppointments = [];
+
+    const startOfToday = moment()
+      .startOf('day')
+      .toISOString();
+
+    try {
+      if (environment.isLocalhost()) {
+        vaAppointments = MOCK_VA_APPOINTENTS;
+        ccAppointments = MOCK_CC_APPOINTMENTS;
+      } else {
+        vaAppointments = await apiRequest(
+          `/vaos/v0/appointments?start_date=${startOfToday}&type=va`,
+        );
+        ccAppointments = await apiRequest(
+          `/vaos/v0/appointments?start_date=${startOfToday}&type=cc`,
+        );
+      }
+      const facilityIDs = uniq(
+        map(
+          vaAppointments,
+          appointment =>
+            getStagingID(appointment?.attributes?.sta6aid)
+              ? `vha_${getStagingID(appointment?.attributes?.sta6aid)}`
+              : undefined,
+        ),
+      );
+
+      if (environment.isLocalhost()) {
```

Comment:
same here. I doubt we'll need to do much mocking of this API in Cypress, but ideally we should be able to easily mock a failed call to make sure we handle that situation correctly.